coremltools/converters/mil/frontend/torch/internal_graph.py

```diff
@@ -518,58 +519,22 @@ def cache_model_hierarchy_block(block):
 
     @classmethod
     def from_exir(cls, exir):
-        # exir: torch.export.ExportedProgram
-        exported_program = exir
-
-        nodes = []
-        params = {}
-        outputs = []
-        inputs = OrderedDict(
-            [
-                (i.name, i)
-                for i in extract_inputs_from_exir_program(exported_program=exported_program)
-            ]
+        exported_program: torch.export.ExportedProgram = exir
+        user_inputs, lifted_parameters, lifted_buffers, lifted_constants = (
+            extract_inputs_from_exir_program(exported_program)
         )
 
-        inputs_to_parameters = exported_program.graph_signature.inputs_to_parameters
-        inputs_to_buffers = exported_program.graph_signature.inputs_to_buffers
-
-        inputs_to_consts = {**inputs_to_parameters, **inputs_to_buffers}
-        consts_to_inputs = {
-            v: k if not k.startswith("%") else k[1:] for k, v in inputs_to_consts.items()
-        }
-
-        # Add params
-        for name, param in exported_program.state_dict.items():
-            if not isinstance(param, torch.Tensor):
-                raise NotImplementedError(
-                    f"For ExecuTorch paramter, only support torch.Tensor, but got {type(param)}"
-                )
-            params[name if name not in consts_to_inputs else consts_to_inputs[name]] = param
-        # Non-persistent buffers may be missing from state_dict, but we still need their values
-        # Reference: https://github.com/pytorch/executorch/pull/1802
-        for name, buffer in zip(exported_program.graph_signature.buffers, exported_program.buffers()):
-            if not isinstance(buffer, torch.Tensor):
-                raise NotImplementedError(
-                    f"For ExecuTorch buffer, only support torch.Tensor, but got {type(buffer)}"
-                )
-            params_name = consts_to_inputs[name]
-            if params_name in params:
-                assert torch.equal(params[params_name], buffer)
-            else:
-                params[params_name] = buffer
-
-        graph_module = exported_program.graph_module
-        graph = graph_module.graph
+        params = {**lifted_parameters, **lifted_buffers, **lifted_constants}
```

Comment:
Do we use these three dictionaries:
1. lifted_parameters
2. lifted_buffers
3. lifted_constant
Seperately?
The only place they are used is to construct the `params`,
then why not they are returned as a single dictionary in ` extract_inputs_from_exir_program`?
Please correct me if I miss anything ...

Reply:
They may be used differently in near future. ExecuTorch treat them as 3 different types rather than a single type for a good reason